src/api/materials.py

```
@@ -0,0 +1,541 @@
+    Cloudflare R2 supports ``createPresignedUrl`` on the bucket binding.
+    If the binding does not expose that method (e.g. older SDK or unit
+    tests), we fall back to a plain ``/api/r2/{key}`` path that the
+    worker can proxy.
+    """
+    try:
+        bucket = _r2_bucket(env)
+        # CF Workers R2 binding: bucket.createPresignedUrl(key, {expiresIn})
+        url = await bucket.createPresignedUrl(key, {"expiresIn": 3600})
+        return str(url)
+    except Exception:
+        pass
```

Ruff (0.16.1)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)